packages/@aws-cdk/aws-rds/lib/instance-engine.ts

```diff
@@ -153,49 +513,246 @@ class PostgresInstanceEngine extends InstanceEngineBase {
+  /** Version "18.0.0.0.ru-2020-01.rur-2020-01.r1". */
+  public static readonly VER_18_0_0_0_2020_01_R1 = OracleEngineVersion.of('18.0.0.0.ru-2020-01.rur-2020-01.r1', '18');
+  /** Version "18.0.0.0.ru-2020-04.rur-2020-04.r1". */
+  public static readonly VER_18_0_0_0_2020_04_R1 = OracleEngineVersion.of('18.0.0.0.ru-2020-04.rur-2020-04.r1', '18');
+
+  /** Version "19" (only a major version, without a specific minor version). */
+  public static readonly VER_19 = OracleEngineVersion.of('19', '19');
+  /** Version "19.0.0.0.ru-2019-07.rur-2019-07.r1". */
+  public static readonly VER_19_0_0_0_2019_07_R1 = OracleEngineVersion.of('19.0.0.0.ru-2019-07.rur-2019-07.r1', '19');
+  /** Version "19.0.0.0.ru-2019-10.rur-2019-10.r1". */
+  public static readonly VER_19_0_0_0_2019_10_R1 = OracleEngineVersion.of('19.0.0.0.ru-2019-10.rur-2019-10.r1', '19');
+  /** Version "19.0.0.0.ru-2020-01.rur-2020-01.r1". */
+  public static readonly VER_19_0_0_0_2020_01_R1 = OracleEngineVersion.of('19.0.0.0.ru-2020-01.rur-2020-01.r1', '19');
+  /** Version "19.0.0.0.ru-2020-04.rur-2020-04.r1". */
+  public static readonly VER_19_0_0_0_2020_04_R1 = OracleEngineVersion.of('19.0.0.0.ru-2020-04.rur-2020-04.r1', '19');
+
+  /**
+   * Creates a new OracleEngineVersion with an arbitrary version.
+   *
+   * @param oracleFullVersion the full version string,
+   *   for example "19.0.0.0.ru-2019-10.rur-2019-10.r1"
+   * @param oracleMajorVersion the major version of the engine,
+   *   for example "19"
+   */
+  public static of(oracleFullVersion: string, oracleMajorVersion: string): OracleEngineVersion {
+    return new OracleEngineVersion(oracleFullVersion, oracleMajorVersion);
+  }
+
+  /** The full version string, for example, "19.0.0.0.ru-2019-10.rur-2019-10.r1". */
+  public readonly oracleFullVersion: string;
+  /** The major version of the engine, for example, "19". */
+  public readonly oracleMajorVersion: string;
+
+  private constructor(oracleFullVersion: string, oracleMajorVersion: string) {
+    this.oracleFullVersion = oracleFullVersion;
+    this.oracleMajorVersion = oracleMajorVersion;
+  }
+}
+
+interface OracleInstanceEngineBaseProps {
   readonly engineType: string;
-  readonly parameterGroupFamilies?: ParameterGroupFamilyMapping[];
+  readonly version?: OracleEngineVersion;
 }
 
-abstract class OracleInstanceEngine extends InstanceEngineBase {
-  constructor(props: OracleInstanceEngineProps) {
+abstract class OracleInstanceEngineBase extends InstanceEngineBase {
+  constructor(props: OracleInstanceEngineBaseProps) {
     super({
       ...props,
       singleUserRotationApplication: secretsmanager.SecretRotationApplication.ORACLE_ROTATION_SINGLE_USER,
       multiUserRotationApplication: secretsmanager.SecretRotationApplication.ORACLE_ROTATION_MULTI_USER,
+      version: props.version
+        ? {
+          fullVersion: props.version.oracleFullVersion,
+          majorVersion: props.version.oracleMajorVersion,
+        }
```

Comment:
`~` _(Nit)_ - Seeing this block over and over again makes me wonder if you could have done this in the InstanceEngineBase if all of the versions had a consistent naming (e.g., `fullVersion` vs `oracleFullVersion`).

Reply:
Ah. This is done on purpose 🙂.

If they all had the same properties (`fullVersion` and `majorVersion`), thanks to TypeScript's structural typing, it would be valid code to, for example, pass an `OracleEngineVersion` where an `SqlServerEngineVersion` is expected (because both classes would "fit" each other, as they had the same set of properties). I want that to be a compilation error, hence why the names are different.